src/controller/task/sweep.go

```diff
@@ -0,0 +1,143 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package task
+
+import (
+	"context"
+
+	"github.com/goharbor/harbor/src/jobservice/job"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/lib/q"
+	"github.com/goharbor/harbor/src/pkg/scheduler"
+	"github.com/goharbor/harbor/src/pkg/task"
+)
+
+var (
+	// SweepCtl is the global sweep controller
+	SweepCtl = NewSweepController()
+)
+
+type SweepParams struct {
+	// ExecRetainCounts records the retained execution counts for different vendor type
+	ExecRetainCounts map[string]int64
+}
+
+const (
+	// SchedulerCallback ...
+	SchedulerCallback = "EXECUTION_SWEEP_CALLBACK"
+	// systemVendorID represents the id for system job.
+	systemVendorID = -1
+
+	cronTypeCustom = "Custom"
+	// run for every hour
+	cronSpec = "0 0 * * * *"
+)
+
+func init() {
+	err := scheduler.RegisterCallbackFunc(SchedulerCallback, sweepCallback)
+	if err != nil {
+		log.Fatalf("failed to register execution sweep job callback, error: %v", err)
+	}
+}
+
+func sweepCallback(ctx context.Context, p string) error {
+	params := &SweepParams{ExecRetainCounts: job.GetExecutionSweeperCount()}
```

Comment:
Should we pass the sweeper count through a parameter? it is either a user configuration or job service configuration and can be retrieved by config in each job task.

Reply:
Retrieve from job is also an option, do we need to consider exposing the configuration to outside for user?